horreum-backend/src/test/java/io/hyperfoil/tools/horreum/svc/RunServiceTest.java

```diff
@@ -58,32 +66,222 @@ public class RunServiceTest extends BaseServiceTest {
+      BlockingQueue<DataSet> dataSetQueue = eventConsumerQueue(DataSet.class, DataSet.EVENT_NEW);
+      Schema schema = createExampleSchema(info, false);
+
+      assertNotNull(schema.id);
+      NamedJsonPath path = null; // no extractors
+      Transformer transformer = createExampleTransformerWithJsonPath("acme", schema, "", path);
+      Test exampleTest = createExampleTest(getTestName(info));
+      Test test = createTest(exampleTest);
+      addTransformer(test, transformer);
+      dataSetQueue.drainTo(new ArrayList<>());
+      uploadRun(runWithValue(schema, 42.0d), test.name);
+
+      DataSet event = dataSetQueue.poll(POLL_DURATION_SECONDS, TimeUnit.SECONDS);
+      assertNotNull(event);
+      JsonNode node = event.data;
+      assertTrue(node.isArray());
+      assertEquals(1, node.size());
+      assertEquals(schema.uri, node.path(0).path("$schema").textValue());
+   }
+
+   @org.junit.jupiter.api.Test
+   public void testTransformationWithNestedSchema(TestInfo info) throws InterruptedException {
+      BlockingQueue<DataSet> dataSetQueue = eventConsumerQueue(DataSet.class, DataSet.EVENT_NEW);
+      Schema acmeSchema = createExampleSchema("AcmeCorp", "AcmeInc", "AcmeRrUs", false);
+      Schema roadRunnerSchema = createExampleSchema("RoadRunnerCorp", "RoadRunnerInc", "RoadRunnerRrUs", false);
+
+      NamedJsonPath acmePath = createExampleNamedJsonPath("foo", "$.value");
+      String acmeFunction = createFunction("foo");
+      Transformer acmeTransformer = createExampleTransformerWithJsonPath("acme", acmeSchema, acmeFunction, acmePath);
+      NamedJsonPath roadRunnerPath = createExampleNamedJsonPath("bah", "$.value");
+      String bahFunction = createFunction("bah");
+      Transformer roadRunnerTransformer = createExampleTransformerWithJsonPath("roadrunner", roadRunnerSchema, bahFunction, roadRunnerPath);
+
+      Test exampleTest = createExampleTest(getTestName(info));
+      Test test = createTest(exampleTest);
+      addTransformer(test, acmeTransformer, roadRunnerTransformer);
+
+      dataSetQueue.drainTo(new ArrayList<>());
+      int runId = uploadRun(runWithValue(42.0d, acmeSchema, roadRunnerSchema), test.name);
+
+      DataSet event = dataSetQueue.poll(POLL_DURATION_SECONDS, TimeUnit.SECONDS);
+
+      assertNotNull(event);
+      JsonNode node = event.data;
+      assertTrue(node.isArray());
+      assertEquals(2 , node.size());
+      JsonNode acme = node.path(0).path("outcome").path("foo");
+      validate( "42", acme);
+      JsonNode runner = node.path(1).path("outcome").path("bah");
+      validate( "42", runner);
+      try (CloseMe ignored = roleManager.withRoles(em, Arrays.asList(TESTER_ROLES[2]))) {
+         Run run = Run.findById(runId);
+         assertEquals(1, run.datasets.size());
       }
    }
 
+   @org.junit.jupiter.api.Test
+   public void testTransformationSingleSchemaTestWithoutTransformer(TestInfo info) throws InterruptedException {
```

Comment:
We need a test that verifies that without any transformer,
```
{
   "$schema": "A",
   "value": 1,
   "nestedB": {
     "$schema": "B"
     "value": 2
   },
   "nestedC": {
     "$schema": "C"
     "value": 3
   }
}
```
gets converted into
 ```
[ {
  "$schema": "A",
  "value": 1,
}, {  
  "$schema": "B"
  "value": 2
}, {
  "$schema": "C"
  "value": 3
} ]
```
(Actually, ATM it is fine if "A" keeps the nested fields inside, but they need to be extracted out as well).

Reply:
Added in the next commit.